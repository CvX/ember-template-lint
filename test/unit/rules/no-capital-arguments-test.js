'use strict';

const generateRuleTests = require('../../helpers/rule-test-harness');

generateRuleTests({
  name: 'no-capital-arguments',

  config: true,

  good: ['<Foo @name="bar" />', '@foo'],

  bad: [
    {
      template: '<Foo @Name="bar" />',
      verifyResults(results) {
        expect(results).toMatchInlineSnapshot(`
          Array [
            Object {
              "column": 5,
              "endColumn": 16,
              "endLine": 1,
              "filePath": "layout.hbs",
              "line": 1,
              "message": "Capital argument names is not supported",
              "rule": "no-capital-arguments",
              "severity": 2,
              "source": "Name",
            },
          ]
        `);
      },
    },
    {
      template: '<Foo @_ame="bar" />',
      verifyResults(results) {
        expect(results).toMatchInlineSnapshot(`
          Array [
            Object {
              "column": 5,
              "endColumn": 16,
              "endLine": 1,
              "filePath": "layout.hbs",
              "line": 1,
              "message": "Capital argument names is not supported",
              "rule": "no-capital-arguments",
              "severity": 2,
              "source": "_ame",
            },
          ]
        `);
      },
    },
    {
      template: '{{@Name}}',
      verifyResults(results) {
        expect(results).toMatchInlineSnapshot(`
          Array [
            Object {
              "column": 3,
              "endColumn": 7,
              "endLine": 1,
              "filePath": "layout.hbs",
              "line": 1,
              "message": "Capital argument names is not supported",
              "rule": "no-capital-arguments",
              "severity": 2,
              "source": "Name",
            },
          ]
        `);
      },
    },
    {
      template: '{{@_Name}}',
      verifyResults(results) {
        expect(results).toMatchInlineSnapshot(`
          Array [
            Object {
              "column": 3,
              "endColumn": 8,
              "endLine": 1,
              "filePath": "layout.hbs",
              "line": 1,
              "message": "Capital argument names is not supported",
              "rule": "no-capital-arguments",
              "severity": 2,
              "source": "_Name",
            },
          ]
        `);
      },
    },

    ...['@arguments', '@args', '@block', '@else'].map((el) => {
      return {
        template: `{{${el}}}`,
        verifyResults(results) {
          expect(results).toMatchSnapshot();
        },
      };
    }),

    ...['@arguments', '@args', '@block', '@else'].map((el) => {
      return {
        template: `<MyComponent ${el}={{42}} />`,
        verifyResults(results) {
          expect(results).toMatchSnapshot();
        },
      };
    }),
  ],
});
