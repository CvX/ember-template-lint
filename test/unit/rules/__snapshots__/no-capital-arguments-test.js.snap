// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no-capital-arguments <MyComponent @args={{42}} />: logs errors 1`] = `
Array [
  Object {
    "column": 13,
    "endColumn": 25,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "@args is reserved argument name, try to use another",
    "rule": "no-capital-arguments",
    "severity": 2,
    "source": "args",
  },
]
`;

exports[`no-capital-arguments <MyComponent @arguments={{42}} />: logs errors 1`] = `
Array [
  Object {
    "column": 13,
    "endColumn": 30,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "@arguments is reserved argument name, try to use another",
    "rule": "no-capital-arguments",
    "severity": 2,
    "source": "arguments",
  },
]
`;

exports[`no-capital-arguments <MyComponent @block={{42}} />: logs errors 1`] = `
Array [
  Object {
    "column": 13,
    "endColumn": 26,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "@block is reserved argument name, try to use another",
    "rule": "no-capital-arguments",
    "severity": 2,
    "source": "block",
  },
]
`;

exports[`no-capital-arguments <MyComponent @else={{42}} />: logs errors 1`] = `
Array [
  Object {
    "column": 13,
    "endColumn": 25,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "@else is reserved argument name, try to use another",
    "rule": "no-capital-arguments",
    "severity": 2,
    "source": "else",
  },
]
`;

exports[`no-capital-arguments {{@args}}: logs errors 1`] = `
Array [
  Object {
    "column": 3,
    "endColumn": 7,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "@args is reserved argument name, try to use another",
    "rule": "no-capital-arguments",
    "severity": 2,
    "source": "args",
  },
]
`;

exports[`no-capital-arguments {{@arguments}}: logs errors 1`] = `
Array [
  Object {
    "column": 3,
    "endColumn": 12,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "@arguments is reserved argument name, try to use another",
    "rule": "no-capital-arguments",
    "severity": 2,
    "source": "arguments",
  },
]
`;

exports[`no-capital-arguments {{@block}}: logs errors 1`] = `
Array [
  Object {
    "column": 3,
    "endColumn": 8,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "@block is reserved argument name, try to use another",
    "rule": "no-capital-arguments",
    "severity": 2,
    "source": "block",
  },
]
`;

exports[`no-capital-arguments {{@else}}: logs errors 1`] = `
Array [
  Object {
    "column": 3,
    "endColumn": 7,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "@else is reserved argument name, try to use another",
    "rule": "no-capital-arguments",
    "severity": 2,
    "source": "else",
  },
]
`;
